diff --git a/src/meson.build b/src/meson.build
index 535338d..53affee 100644
--- a/src/meson.build
+++ b/src/meson.build
@@ -29,9 +29,13 @@ conf.set_quoted('VERSION', version)
 conf.set_quoted('LOCALEDIR', get_option('prefix') / get_option('localedir'))
 configure_file(output: 'config.h', configuration: conf)
 
-sources += gnome.compile_resources('g4music-resources',
+pkgdatadir = get_option('prefix') / get_option('datadir') / meson.project_name()
+
+gnome.compile_resources('g4music-resources',
   'g4music.gresource.xml',
-  c_name: 'g4music'
+  c_name: 'g4music',
+  gresource_bundle: true,
+  install: true
 )
 
 executable('g4music', sources,
